From 7526b77d6866f48ed59d6a79a2ceb821d8c3618d Mon Sep 17 00:00:00 2001
From: Jonathan Hamilton <jonathan.hamilton@imgtec.com>
Date: Thu, 17 Jul 2014 15:54:44 -0700
Subject: [PATCH 402/634] video: adf: Cleanup sw_sync timeline at
 adf_device_destroy

If a sw_sync timeline was created by ADF (for drivers that do not implement
ops->complete_fence) we should clean it up when the ADF device is
destroyed.

Change-Id: Idd90180fcae56a87111f7d12bdd80190756a6b80
Signed-off-by: Jonathan Hamilton <jonathan.hamilton@imgtec.com>
---
 drivers/video/adf/adf.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/video/adf/adf.c b/drivers/video/adf/adf.c
index 231881c..42c30c0 100644
--- a/drivers/video/adf/adf.c
+++ b/drivers/video/adf/adf.c
@@ -613,6 +613,10 @@ void adf_device_destroy(struct adf_device *dev)
 	}
 	mutex_destroy(&dev->post_lock);
 	mutex_destroy(&dev->client_lock);
+
+	if (dev->timeline)
+		sync_timeline_destroy(&dev->timeline->obj);
+
 	adf_obj_destroy(&dev->base, &adf_devices);
 }
 EXPORT_SYMBOL(adf_device_destroy);
-- 
1.9.1

